solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetAccessStatusFilter.java

```diff
@@ -105,13 +108,45 @@ public void setObjectAclFactory(ObjectAclFactory oaf) {
             status.add(FacetConstants.PUBLIC_ACCESS);
         }
 
-        if (objPrincRoles.size() > 0) {
+        if (addPatronSettings(dip.getContentObject())) {
             status.add(FacetConstants.PATRON_SETTINGS);
         }
 
         return status;
     }
 
+    private boolean addPatronSettings(ContentObject contentObj) {
```

Comment:
I'd suggest giving the method a name that indicates that it is evaluating a condition rather than actually performing an addition. Maybe `needsPatronSettingsStatus`, or merge it with `hasPatronSettings`